### PR TITLE
Remove as much as std::function as possible.

### PR DESCRIPTION
**Pull request by verri**
Part of the saga "Removal of all useless heap allocation".

### CODE REVIEW

**Comment by verri** [comment]
@JorgeLuizFranco , let's test something.

Please review the code and approve.

I think a rule that one person should review before merging might be appropriate for our workflow.

File: include/uat/simulation.hpp
Comment (JorgeLuizFranco) [inline include/uat/simulation.hpp:1]: Removing `acessor` made it much cleaner.

File: include/uat/simulation.hpp
Comment (JorgeLuizFranco) [inline include/uat/simulation.hpp:20]: I think `factory_fn` was a better name for what it does.

File: include/uat/simulation.hpp
Comment (verri) [reply include/uat/simulation.hpp:1]: Great! It was not needed anymore once the user receives only a const version of the structure.

**Review by JorgeLuizFranco — APPROVED** [review]

**Comment by verri** [comment]
Fixes #6

File: include/uat/simulation.hpp
Comment (verri) [reply include/uat/simulation.hpp:20]: This was a hard choice. Factory is actually an object, not a reference. And can become much more complex in the future, such as having more methods. That was the reason I chose this naming.

**Pull request merged**
